### PR TITLE
Implement food 

### PR DESCRIPTION
**Pull request by kamcio96**
All items and theirs effects

Implemented events:
- PlayerItemConsumeEvent
- FoodLevelChangeEvent
- EntityRegainHealthEvent

Something more?

### CODE REVIEW

File: src/main/java/net/glowstone/block/itemtype/ItemFood.java [hunk 1]
```diff
@@ -1,5 +1,42 @@
...
+        this.saturation = saturation;
+    }
+
+    public boolean eat(GlowPlayer player, ItemStack item) {
+
+        FoodLevelChangeEvent foodLevelChangeEvent = new FoodLevelChangeEvent(player, foodLevel + player.getFoodLevel());
+        EventFactory.callEvent(foodLevelChangeEvent);
+        if (foodLevelChangeEvent.isCancelled()) return false;
+
+        player.setFoodLevelAndSaturdation(foodLevelChangeEvent.getFoodLevel(), saturation);
```
Comment (mastercoms) [inline src/main/java/net/glowstone/block/itemtype/ItemFood.java]: Saturation

File: src/main/java/net/glowstone/block/itemtype/ItemTimedUsage.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+public class ItemTimedUsage extends ItemType {
+
+    @Override
+    public void rightClickAir(GlowPlayer player, ItemStack holding) {
+        startUse(player, holding);
+    }
+
+    @Override
+    public void rightClickBlock(GlowPlayer player, GlowBlock target, BlockFace face, ItemStack holding, Vector clickedLoc) {
```
Comment (mastercoms) [inline src/main/java/net/glowstone/block/itemtype/ItemTimedUsage.java]: Check if block can be interacted with first?

File: src/main/java/net/glowstone/entity/GlowPlayer.java [hunk 3]
```diff
@@ -1319,6 +1390,15 @@ public void setFoodLevel(int food) {
         sendHealth();
     }
 
+    // todo: effects
+    // todo: swim
+    // todo: jump
+    // todo: food poisioning
+    // todo: hump and sprint
```
Comment (mastercoms) [inline src/main/java/net/glowstone/entity/GlowPlayer.java]: what is hump?

Comment (kamcio96) [inline src/main/java/net/glowstone/entity/GlowPlayer.java]: jump* XD

File: src/main/java/net/glowstone/block/itemtype/ItemFood.java (hunk 1 above)
Comment (kamcio96) [inline src/main/java/net/glowstone/block/itemtype/ItemFood.java]: Why lower char?

**Comment by kamcio96** [comment]
Sorry for pom changes :/ I used auto reformat code

File: src/main/java/net/glowstone/block/itemtype/ItemFish349.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
+package net.glowstone.block.itemtype;
+
+import net.glowstone.entity.GlowPlayer;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.potion.PotionEffectType;
+
+public class ItemFish349 extends ItemFood {
```
Comment (deathcap) [inline src/main/java/net/glowstone/block/itemtype/ItemFish349.java]: Class name could be more descriptive (not including magic numbers), how about ItemFishRaw?

File: src/main/java/net/glowstone/block/itemtype/ItemFish350.java [hunk 5]
```diff
@@ -0,0 +1,30 @@
+package net.glowstone.block.itemtype;
+
+import org.bukkit.inventory.ItemStack;
+
+public class ItemFish350 extends ItemFood {
```
Comment (deathcap) [inline src/main/java/net/glowstone/block/itemtype/ItemFish350.java]: ItemFishCooked?

File: pom.xml [hunk 6]
```diff
@@ -1,4 +1,5 @@
-<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
```
Comment (deathcap) [inline pom.xml]: Please remove the unnecessary formatting changes (to pom.xml and a few other files below). FYI you can see all your changes in the PR from the command-line with `git diff master`.

File: src/test/java/net/glowstone/constants/AchievementTest.java [hunk 7]
```diff
@@ -8,7 +8,7 @@
 
 import java.util.Collection;
 
-import static org.junit.Assert.*;
+import static org.junit.Assert.assertTrue;
```
Comment (deathcap) [inline src/test/java/net/glowstone/constants/AchievementTest.java]: Unrelated/unnecessary reformatting change (and all other files below)

**Comment by kamcio96** [comment]
Updated

**Comment by mastercoms** [comment]
Nice work so far, but the eating time is a bit too long, and cooked porkchops don't feed you. That's all I got from the limited time I had to test, I'll have to test more things later, when I have time.

**Comment by kamcio96** [comment]
I couldn't find on wiki how much time it should take.

**Comment by mastercoms** [comment]
play around with the values until the animation and server response are aligned.

**Comment by mastercoms** [comment]
Cooked porkchops still do not work.
